platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -299,7 +304,7 @@ def get_onie_firmware_info(self, image_path):
 
         return firmware_info
 
-    def update_firmware(self, image_path):
+    def update_firmware(self, image_path, reboot):
```

Comment:
@alexrallen suggest to define a separate API `install_firmware_no_reboot`

Reply:
After reviewing this and what would be the cleanest solution with the least impact is to update the "install firmware" (for the components) and the "update firmware" (for the ONIEUpdater class) methods to include named parameters for specifying reboots with defaults set for the original behavior....

So in this case (ONIEUpdater) the function signature would look like....
`def update_firmware(self, image_path, allow_reboot=True)`

and in the case of components (ComponentSSD / ComponentONIE / etc.) the signature would look like...
`def install_firmware(self, image_path, allow_reboot=True)`

This will allow these methods to: 
* Comply with the abstract API defined by the SONiC platform API spec
* Perverse 100% backwards compatibility
* Allows calling them from new contexts that don't support immediate reboots

Without me having to define new methods which are extremely simply duplicates of the existing methods which I feel would adversely impact the maintainability and readability of the code.

Reply:
> So in this case (ONIEUpdater) the function signature would look like....
def update_firmware(self, image_path, allow_reboot=True)

@alexrallen looks like the signature within the code is different ...

Reply:
Updated to be consistent